bindings/go/oci/ctf/store.go

```diff
@@ -152,11 +154,15 @@ func (s *Repository) Resolve(ctx context.Context, reference string) (ociImageSpe
 
 	repo := s.repo
 
-	if prefix := wellKnownRegistryCTF + "/"; strings.HasPrefix(reference, prefix) {
-		reference = strings.TrimPrefix(reference, prefix)
-	}
-	if prefix := repo + ":"; strings.HasPrefix(reference, prefix) {
-		reference = strings.TrimPrefix(reference, prefix)
+	// if we do not have a pure digest, we need to parse the reference
+	// loosely because it could be that registry/repository information is prefixed to the actual reference.
+	if _, err := digest.Parse(reference); err != nil {
+		if idx := strings.Index(reference, ":"); idx != -1 {
+			reference = reference[idx+1:]
+		}
+		if idx := strings.Index(reference, "@"); idx != -1 {
+			reference = reference[:idx]
+		}
```

Comment:
Doesn't this _remove_ the digest part? Is that what you want?

Reply:
hmm youre right im currently not discovering a failed resolve because the tag is still resolvable, i want it to be the other way around. good catch!

Reply:
👁️ 👄 👁️

Reply:
This will now fail anything that has a port in. For example `registry:5000/repo:tag`, right?

Reply:
Not particularly fail I guess, so `registry:5000/repo:tag` will just become `tag`.

Reply:
No, actually, Index is the first occurrence. So this would become `5000/repo:tag`.

Reply:
Nice, now it looks good, I think.